sdcflows/transform.py

```diff
@@ -215,36 +217,108 @@ def to_displacements(self, ro_time, pe_dir, itk_format=True):
             A NIfTI 1.0 object containing the distortion.
 
         """
-        # Set polarity & scale VSM (voxel-shift-map) by readout time
-        vsm = self.shifts.get_fdata().copy()
-        pe_axis = "ijk".index(pe_dir[0])
-        vsm *= -1.0 if pe_dir.endswith("-") else 1.0
-        vsm *= ro_time
-
-        # Shape of displacements field
-        # Note that ITK NIfTI fields are 5D (have an empty 4th dimension)
-        fieldshape = tuple(list(vsm.shape[:3]) + [1, 3])
-
-        # Convert VSM to voxel displacements
-        ijk_deltas = np.zeros((vsm.size, 3), dtype="float32")
-        ijk_deltas[:, pe_axis] = vsm.reshape(-1)
-
-        # To convert from VSM to RAS field we just apply the affine
-        aff = self.shifts.affine.copy()
-        aff[:3, 3] = 0  # Translations MUST NOT be applied, though.
-        xyz_deltas = nb.affines.apply_affine(aff, ijk_deltas)
-        if itk_format:
-            # ITK displacement vectors are in LPS orientation
-            xyz_deltas[..., (0, 1)] *= -1.0
-
-        xyz_nii = nb.Nifti1Image(
-            xyz_deltas.reshape(fieldshape),
-            self.shifts.affine,
-            None,
-        )
-        xyz_nii.header.set_intent("vector", (), "")
-        xyz_nii.header.set_xyzt_units("mm")
-        return xyz_nii
+        return fmap_to_disp(self.shifts, ro_time, pe_dir, itk_format=itk_format)
+
+
+def fmap_to_disp(fmap_nii, ro_time, pe_dir, itk_format=True):
+    """
+    Convert a fieldmap in Hz into an ITK/ANTs-compatible displacements field.
+
+    The displacements field can be calculated following
+    `Eq. (2) in the fieldmap fitting section
+    <sdcflows.workflows.fit.fieldmap.html#mjx-eqn-eq%3Afieldmap-2>`__.
+
+    Parameters
+    ----------
+    fmap_nii : :obj:`os.pathlike`
+        Path to a voxel-shift-map (VSM) in NIfTI format
+    ro_time : :obj:`float`
+        The total readout time in seconds (only if ``vsm=False``).
+    pe_dir : :obj:`str`
+        The ``PhaseEncodingDirection`` metadata value (only if ``vsm=False``).
```

Comment:
I think the `vsm=False` comments are out-of-place here.
```suggestion
        The total readout time in seconds
    pe_dir : :obj:`str`
        The ``PhaseEncodingDirection`` metadata value
```